### PR TITLE
GK parser

### PR DESCRIPTION
**Pull request by imkiva**
Goodbye `AyaProducer`. We will miss you.

- [x] Implement GK parser and producer
- [x] Migrate to GK parser everywhere (except REPL)
- [x] Report error elements

### CODE REVIEW

File: parser-migrate/src/main/java/org/aya/parser/ij/AyaFleetParser.java [hunk 1]
```diff
@@ -0,0 +1,24 @@
+package org.aya.parser.ij;
+
+import com.intellij.psi.FleetPsiParser;
+import com.intellij.psi.builder.FleetPsiBuilder;
+import com.intellij.psi.tree.IFileElementType;
+import org.aya.parser.ij.parser.AyaParserDefinitionBase;
+import org.jetbrains.annotations.NotNull;
+
+public class AyaFleetParser extends FleetPsiParser.DefaultPsiParser {
```
[the remark this answers is not in the document]
Comment (imkiva) [reply parser-migrate/src/main/java/org/aya/parser/ij/AyaFleetParser.java:9]: It parses text to `GenericNode`

[the remark this answers is not in the document]
Comment (ice1000) [reply parser-migrate/src/main/java/org/aya/parser/ij/AyaFleetParser.java:9]: Needed by Psi factory?

[the remark this answers is not in the document]
Comment (imkiva) [reply parser-migrate/src/main/java/org/aya/parser/ij/AyaFleetParser.java]: See [AyaGKParserImpl.java](https://github.com/aya-prover/aya-dev/blob/7af67f8921355a0bd9554dc283f699ceb786dfcc/cli/src/main/java/org/aya/cli/parse/AyaGKParserImpl.java)

**Pull request merged**